packages/slice-machine/lib/env/manifest.ts

```diff
@@ -125,8 +145,12 @@ function handleManifest(cwd: string): ManifestInfo {
       Models.Manifest.decode(json),
       fold(
         // failure handler
-        () => {
-          throw new Error();
+        (errors) => {
```

Comment:
You should not make some extra validation after using a codec otherwise it means that your validator is not accurate and you have two successive layers of validation. All the validation of `validate` should not be part of your codec.
should be something like:
```
getOrElseW(
  (errors) => {
    const messages = formatValidationErrors(errors);
    messages.forEach((error) => {
      console.error("[sm.json] " + error);
    });
    return undefined // or encode error
  }
)(Models.Manifest.decode(json))
```

And here you still cast + validate in the error handler, you cast a type that was rejected by the codec so are sure already that the json doesn't match the type that you cast on it.

Reply:
It's not me, this is the code was already there. 

I did try to remove that validation function, but here's a few issues.

+ in some places in the code framework is needed, in others it's not.

Reply:
The only concern is the model, is the framework required for sm.json? In any case that should be part of the codec. Optional or not if it's there it should be properly validated as a framework but if the code requires it in some places it means that it's required for your app to function properly